scripts/deploy.js

```diff
@@ -4,26 +4,63 @@
 // When running the script with `hardhat run <script>` you'll find the Hardhat
 // Runtime Environment's members available in the global scope.
 const hre = require("hardhat");
+const delay = s => new Promise(res => setTimeout(res, s*1000));
 
 async function main() {
   await hre.run('compile');
 
   // We get the contract to deploy
   const dappStakingPoolFactory = await hre.ethers.getContractFactory("DappStakingPool");
-  const dappTokenFactory = await hre.ethers.getContractFactory("DappToken");
+  const funderFactory = await hre.ethers.getContractFactory("Funder");
 
-  const dappTokenContract = await dappTokenFactory.deploy();
-  await dappTokenContract.deployed();
-  console.log("Dapp Token deployed to:", dappTokenContract.address);
+  const dappStakingPoolProxy = await upgrades.deployProxy(dappStakingPoolFactory, [
+    "0x853c2D147a1BD7edA8FE0f58fb3C5294dB07220e", // liquidity protection
+    "0xf5FAB5DBD2f3bf675dE4cB76517d4767013cfB55", // liq protection store
+    "0x33A23d447De16a8Ff802c9Fcc917465Df01A3977", // dapp bnt anchor
+    "0x939b462ee3311f8926c047d2b576c389092b1649", // dapp token
+    "0x1f573d6fb3f13d689ff844b4ce37794d79a7ff1c", // bnt token
+    await hre.ethers.provider.getBlockNumber(), // start block
+    205000 // 20.5 DAPPs per block with 4 decimal precision
+  ]);
+  await dappStakingPoolFactory.attach(dappStakingPoolProxy.address);
 
-  const dappStakingPoolContract = await dappStakingPoolFactory.deploy();
-  await dappStakingPoolContract.deployed();
-  console.log("Dapp Staking Pool deployed to:", dappStakingPoolContract.address);
+  // start at 0% for rewards so all goes to IL
+  const funderProxy = await upgrades.deployProxy(funderFactory, [dappStakingPoolProxy.address,"0x939b462ee3311f8926c047d2b576c389092b1649",0]);
+  await funderFactory.attach(funderProxy.address);
 
+  const gnosisSafe = '0x5288d36112fe21be1a24b236be887C90c3AE7090';
+
+  console.log("Transferring ownership of Pool & Funder...");
+  await dappStakingPoolProxy.transferOwnership(gnosisSafe);
+  await funderProxy.transferOwnership(gnosisSafe);
+  console.log("Transferred ownership of Pool & Funder to:", gnosisSafe);
+
+  console.log("Transferring ownership of ProxyAdmin...");
+  // The owner of the ProxyAdmin can upgrade our contracts
+  await upgrades.admin.transferProxyAdminOwnership(gnosisSafe);
+  console.log("Transferred ownership of ProxyAdmin to:", gnosisSafe);
```

Comment:
if i'm not mistaken, the gnosis safe must accept ownership as well

Reply:
Not sure how to do that.  The documentation doesn't seem to recommend that or have a function for it.

https://docs.openzeppelin.com/upgrades-plugins/1.x/api-hardhat-upgrades#admin-transfer-proxy-admin-ownership
https://docs.openzeppelin.com/defender/guide-upgrades#transfer-control-of-upgrades-to-a-multisig
https://github.com/OpenZeppelin/openzeppelin-upgrades/blob/c9df3e74993243421d5f850f27589cc8830a7bb9/packages/plugin-truffle/src/admin.ts#L21

Also not seeing the function itself anymore on `Owned.sol`

I don't see the ability to accept on gnosis itself either, though not seeing anything. https://rinkeby.etherscan.io/address/0x5288d36112fe21be1a24b236be887C90c3AE7090#code

Reply:
I think you're right, not sure where I remember seeing this...